doc/plugin-develop.md

```diff
@@ -168,8 +168,8 @@ local _M = {
 Determine which phase to run, generally access or rewrite. If you don't know the [Openresty life cycle](https://openresty-reference.readthedocs.io/en/latest/Directives/), it's
 recommended to know it in advance. For example key-auth is an authentication plugin, thus the authentication should be completed
 before forwarding the request to any upstream service. Therefore, the plugin can be executed in the rewrite and access phases.
-In APISIX, the authentication logic is implemented in the rewrite phase. Generally, IP access and interface
-permission are completed in the access phase.
+In APISIX, the authentication logic is implemented in the access phase. Generally, IP access and interface
```

Comment:
We don't need to change it as authz-keycloak is for authorization.

Reply:
Ok, I can revert this one. However, note that some authentication plugins currently run in the access phase, not the rewrite one. So the document doesn’t reflect the current state correctly, I believe.

Reply:
@jenskeiner
We can fix it later, but in those plugins, not the documentation.

Reply:
@jenskeiner
Thank you for the problem you point out. I just search for all plugins which has `type = 'auth'` (excluding the authz-keycloak) and find out `basic-auth` doesn't run the phase correctly.

Reply:
The reason why we need to run authentication plugins in the rewrite phases:
https://github.com/apache/apisix/blob/f4161d39a495879edffbc75e2099f6e487bae81d/apisix/init.lua#L508-L520
And:
https://github.com/apache/apisix/blob/f4161d39a495879edffbc75e2099f6e487bae81d/apisix/admin/consumers.lua#L50

Plugins has `type = 'auth'` should run in rewrite phase and set up the consumer for current api_ctx.

Reply:
Ok, understood why you need to run in the rewrite phase. Then I would propose to change the phase for all authentication-type plugins to *rewrite*, instead of adjusting the phase for `authz-keycloak` from *rewrite* to *access*. Also, I think it would make the plugin development guidelines clearer if the reason they need to run in the rewrite phase was pointed out more clearly.

If you're fine with this, I can update this PR accordingly. Just let me know what you think.

Reply:
> We don't need to change it as authz-keycloak is for authorization.

Reverted.